src/libs/PersonalDetailsUtils.ts

```diff
@@ -24,9 +24,14 @@ Onyx.connect({
     },
 });
 
+const hiddenText = Localize.translateLocal('common.hidden');
+const substringStartIndex = 8;
 function getDisplayNameOrDefault(passedPersonalDetails?: Partial<PersonalDetails> | null, defaultValue = '', shouldFallbackToHidden = true): string {
-    const displayName = passedPersonalDetails?.displayName ? passedPersonalDetails.displayName.replace(CONST.REGEX.MERGED_ACCOUNT_PREFIX, '') : '';
-    const fallbackValue = shouldFallbackToHidden ? Localize.translateLocal('common.hidden') : '';
+    let displayName = passedPersonalDetails?.displayName ?? '';
+    if (displayName.startsWith(CONST.MERGED_ACCOUNT_PREFIX)) {
+        displayName = displayName.substring(substringStartIndex);
```

Comment:
if I understand we want everything past `MERGED_`, doesn't this return slightly incorrectly without the 1st character right after the `_`?

Reply:
Correct. I missed that some how. Fixed now 👍

Reply:
uff good we spotted that hahah

Reply:
Might be worth documenting the `substringStartIndex` constant with this little explanation! It'd be valuable to have this information in a comment - what it helps to strip and why.

Reply:
Not sure why we need the extra variable, why not just use  `displayName = displayName.substring(CONST.MERGED_ACCOUNT_PREFIX.length);`?

Reply:
bump on my question above @hurali97 please

Reply:
@youssef-lr I added the extra variable just for better readability. I can remove the variable and add the comment instead 👍